vrclient_x64/vrclient_x64/vrclient_main.c

```diff
@@ -891,6 +1008,49 @@ static EVRCompositorError ivrcompositor_submit_vulkan(
     return cpp_func(linux_side, eye, tex, bounds, flags);
 }
 
+EVROverlayError ivroverlay_set_overlay_texture(
+		EVROverlayError (*cpp_func)(void *, VROverlayHandle_t, Texture_t *),
+		void *linux_side, VROverlayHandle_t overlayHandle, Texture_t *texture,
+		unsigned int version, struct overlay_data *user_data)
+{
+	IUnknown *texture_iface;
+	HRESULT hr;
+
+	TRACE("%p, overlayHandle = %#x, texture = %p\n", linux_side, overlayHandle, texture);
+
+	TRACE("texture->eType = [%d] %s\n", texture->eType, vrclient_texture_type_name(texture->eType));
+
+	switch (texture->eType)
+	{
+		case TextureType_DirectX:
+			TRACE("D3D11\n");
+			texture_iface = texture->handle;
+			TRACE("texture_iface = %p\n", texture_iface);
+			if (texture_iface != NULL) {
+				TRACE("texture_iface->lpVtbl = %p\n", texture_iface->lpVtbl);
+			} else {
+				goto overlay_warn_out;
+			}
+
+#ifdef VRCLIENT_HAVE_DXVK
+			{
+				IDXGIVkInteropSurface *dxvk_surface;
+				if (SUCCEEDED(hr = texture_iface->lpVtbl->QueryInterface(texture_iface, &IID_IDXGIVkInteropSurface, (void **)&dxvk_surface))) {
+					return ivroverlay_set_overlay_texture_dxvk(cpp_func, linux_side, overlayHandle, texture, version, user_data, dxvk_surface);
+				}
+			}
+#endif
```

Comment:
We're implicitly falling through to the `TextureType_Vulkan` case, is that intended (and if so, why)?

Reply:
Looks like when I squashed the overlay_warn_out label got moved to the wrong spot. Functionally it’ll work because ever path returns but you’re right that this wasn’t intended

Reply:
Resolved this with an amended commit